Plugins/Flow.Launcher.Plugin.BrowserBookmark/Main.cs

```diff
@@ -113,11 +111,6 @@ public Control CreateSettingPanel()
             return new SettingsControl(_settings);
         }
 
-        public void Save()
-        {
-            _storage.Save();
-        }
-
```

Comment:
why is this removed? settings still needs to be saved right

Reply:
I add a auto save for all registered jsonstorage, so that we don't need to configure isavable for setting, unless they are going to save something else.

Reply:
https://github.com/Flow-Launcher/Flow.Launcher/blob/b37f9f338aee9c2ecd5a50ae47cecd2fa4dae5cb/Flow.Launcher/PublicAPIInstance.cs#L139-L146

Reply:
1. would this not be a doubling up then, with PluginManager.Save() call?

2. also you need to move this SaveAllJsonStorage() call to here right so when Flow's settings window closed you want to save all plugin settings:
https://github.com/Flow-Launcher/Flow.Launcher/blob/9d4af83fef727a38e22daf6e636351f42e3e1c08/Flow.Launcher/ViewModel/SettingWindowViewModel.cs#L85-L97

Reply:
3. do we want to remove ISavable?
4. if it's auto saved, might want to document it somewhere right? for those that develop plugins

Reply:
> would this not be a doubling up then, with PluginManager.Save() call?

I guess that one will call Save from ISavable? So I remove the call to save in exist plugin.

> also you need to move this SaveAllJsonStorage() call to here right so when Flow's settings window closed you want to save all plugin settings:

Yeah you are right.

> if it's auto saved, might want to document it somewhere right? for those that develop plugins

Yeah, I forget to add that into the api description

Reply:
> do we want to remove ISavable?

I believe not. There are other things that may need to save than setting (e.g. the program cache in program plugin).

Reply:
![image](https://user-images.githubusercontent.com/45326534/118120222-1e968200-b422-11eb-9781-0c64202028c1.png)

I guess we don't need to add it to viewmodel save. The api save will be called once Flow is disposed.

Reply:
> > do we want to remove ISavable?
> 
> I believe not. There are other things that may need to save than setting (e.g. the program cache in program plugin).

yeah you are right, lets keep it as we do need to save non-settings, but lets document the Save() method in so it's clear, otherwise devs might use it thinking that's how you save settings.

Reply:
> ![image](https://user-images.githubusercontent.com/45326534/118120222-1e968200-b422-11eb-9781-0c64202028c1.png)
> 
> I guess we don't need to add it to viewmodel save. The api save will be called once Flow is disposed.

We need to because it's better to save when user close the settings window, otherwise what of the program crashes, no settings will be saved.

Reply:
Got you

Reply:
> We need to because it's better to save when user close the settings window, otherwise what if the program crashes, no settings will be saved.

Done

Reply:
can you add SaveAllJsonStorage() call to here so when Flow's settings window closed you want to save all plugin settings:
https://github.com/Flow-Launcher/Flow.Launcher/blob/9d4af83fef727a38e22daf6e636351f42e3e1c08/Flow.Launcher/ViewModel/SettingWindowViewModel.cs#L85-L97

Reply:
I think this method is called in SaveAllJsonStorage, so it is impossible to directly call that method here, or else will become infinity recurse.
The call for 
```c#
PluginManager.Save(); 
_storage.Save(); 
```
do save all plugin setting.

Reply:
yeah i see. hmm how do save plugin settings when user closes settings window? PluginManager.Save() only saves plugin settings where the interface ISavable is applied and _storage.Save() saves Flow's own settings.

Reply:
instead of passing the api to the view, which is not even used, we can and should pass it to the viewmodel, this then allows you to call save plugin settings seperately from the view
https://github.com/Flow-Launcher/Flow.Launcher/blob/1e687a213039ba5f0c2fed42c7ef975a2d854141/Flow.Launcher/SettingWindow.xaml.cs#L28

Reply:
> PluginManager.Save() only saves plugin settings where the interface ISavable is applied and _storage.Save() saves Flow's own settings.

I have modified PluginManager.Save() to include saving plugins settings.

Reply:
```c#
public static void Save()
        {
            foreach (var plugin in AllPlugins)
            {
                var savable = plugin.Plugin as ISavable;
                savable?.Save();
            }
            API.SavePluginSettings();
        }
```